### PR TITLE
Remove usages of query stats for verifying input rows in tests

### LINKED ISSUE #4964: Remove usages of query stats for verifying input rows in tests
Currently `QueryStats` are not guaranteed to be correct (updated) on query completion. We use query stats in `HivePartitioningTest.getProcessedLinesCount`  to get the input positions. 

Right now we do not have a reliable way to get this information, so we should possibly remove these tests.

### PR DESCRIPTION
**Pull request by kokosing**
Remove usages of query stats for verifying input rows in tests

Fixes #4964.

This is a workaround as query stats should be used in such tests to
verify how much data was read to see that partition was pruned from
reading. However, due the fact that currently query stats are not
reliable and may return not up-to-date data it was changed to avoid
intermittent failures.

Now partition pruning is checked in a following way. Two partitions are
loaded. One correct and one with incorrect data. When a SELECT is done
only from correct partition, it is passing as partition prunning has
worked. When SELECT reads all data it fails (or return missing data for text)
due the read from incorrect partition.

### CODE REVIEW

**Comment by kokosing** [comment]
@nileema

**Comment by kokosing** [comment]
cc: @petroav

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java [hunk 1]
```diff
@@ -92,31 +96,15 @@ public void testSelectPartitionedHiveTableDifferentFormats()
...
-                Thread.sleep(GET_PROCESSED_LINES_COUNT_RETRY_SLEEP);
-            }
-            catch (InterruptedException e) {
-            }
-            processedLinesCountAllPartitions = getProcessedLinesCount(sqlStatement, allPartitionsQueryResult);
+        try {
+            // This query should fail or return null values for invalid partition data
+            assertThat(query("SELECT * FROM " + tableNameInDatabase)).containsOnly(row(42, 2), row(null, 1));
+        }
+        catch (QueryExecutionException expectedDueToInvalidPartitionData) {
```
Comment (losipiuk) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java]: Please put some constraint on exception message if possible

**Comment by losipiuk** [comment]
looks good

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java (hunk 1 above)
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java]: each format has its own message, they are totally different. I can assert if exception message is matching any of them but I did not like this.

**Comment by petroav** [comment]
The commit message is a little misleading because the test in `TestTablePartitioningInsertInto` is still using `QueryStats` so you're not really removing all usages as the message implies. Granted that test is quarantined.

Looking at the the test in `TestTablePartitioningInsertInto`, maybe it can also be re-written in the same manner? You can add a corrupt partition and have a case where you successfully `INSERT INTO` by not selecting the corrupt partition and a case where you fail by selecting all partitions.

I agree that the long term solution is to fix `QueryStats` but I think this is a good stop gap in the meantime.

**Comment by kokosing** [comment]
@petroav I updated the commit message. I left  TestTablePartitioningInsertInto untouched to not to spread this stop-gap solution too much.

**Comment by raghavsethi** [comment]
@nileema can you please review? Travis is a sea of red.

**Comment by nileema** [comment]
Looks good. I'll merge this.

**Comment by nileema** [comment]
Merged.

**Pull request closed without merging**